source/core/dictionary.h

```diff
@@ -127,15 +127,13 @@ namespace Slang
 			}
 
 		};
-		template<typename T>
-		inline int GetHashPos(T & key) const
+		inline int GetHashPos(TKey& key) const
```

Comment:
My understanding is that the original intention from @csyonghe was to allow the user to look up entries in a `Dictionary<Key,Val>` with any type that is "compatible" with `Key`. In particular, for cases where you have a dictionary with `String` keys you might want to use an `UnownedStringSlice` for lookup, to avoid heap-allocating the body of a new `String` just for the lookup operation.

All that said, I know that the leniency of these routines has been a pain point in interacting with `Dictionary`, so maybe it isn't worth keeping that level of flexiblity.

Reply:
Fair point. That said the feature is not used in any of the codebase, expands the template param space, and and requires an subtle consistency between the actual key type and the 'lookup' key type/s - in equality and hash generation. That said I can re-add if deemed appropriate.